gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -377,6 +378,13 @@ public void apply(JobListener jobListener, JobContext jobContext)
           jobState.setState(JobState.RunningState.COMMITTED);
           isWorkUnitsEmpty = true;
           return;
+        } else {
+          if (workUnitStream.isSafeToMaterialize()) {
+            this.eventSubmitter.submit(JobEvent.WORK_UNITS_CREATED, "workUnitsCreated",
+                Long.toString(workUnitStream.getMaterializedWorkUnitCollection().size()));
```

Comment:
This call will force work units to be materialized and prevent all optimizations from streaming them (in terms of memory and speed). Ideally, the correct way of doing this is to count the work units as we work on them and emit the event at the end of the run.